src/services/smart-action-form-layout-service.js

```diff
@@ -0,0 +1,86 @@
+function lowerCaseFirstLetter(string) {
+  return string.charAt(0).toLowerCase() + string.slice(1);
+}
+
+const subElementsKey = { Row: 'fields', Page: 'elements' };
+const validLayoutComponents = ['Row', 'Page', 'Separator', 'HtmlBlock'];
+class SmartActionFormLayoutService {
+  static validateLayoutElement(element) {
+    if (!validLayoutComponents.includes(element.component)) throw new Error(`${element.component} is not a valid component. Valid components are ${validLayoutComponents.join(' or ')}`);
+    if (element.component === 'Page') {
+      if (!Array.isArray(element.elements)) {
+        throw new Error('Page components must contain an array of fields or layout elements in property \'elements\'');
+      }
+      if (element.elements.some((innerElement) => innerElement.component === 'Page')) {
+        throw new Error('Pages cannot contain other pages');
+      }
+    }
+    if (element.component === 'Row') {
+      if (!Array.isArray(element.fields)) {
+        throw new Error('Row components must contain an array of fields in property \'fields\'');
+      }
+      if (element.fields.some((field) => field.type === 'Layout')) {
+        throw new Error('Row components can only contain fields');
+      }
+    }
+  }
+
+  static parseLayout(
```

Comment:
This code is kinda a duplicate from Agent Node.js ? Maybe it could be good to add a link to the code in the Node.js agent.